server/hotrod/src/main/scala/org/infinispan/server/hotrod/iteration/IterationManager.scala

```diff
@@ -80,18 +80,23 @@ class DefaultIterationManager(val cacheManager: EmbeddedCacheManager) extends It
 
    override def start(cacheName: String, segments: Option[JavaBitSet], filterConverterFactory: Option[String], batch: Integer): IterationId = {
       val iterationId = UUID.randomUUID().toString
-      val entryRetriever = cacheManager.getCache(cacheName).getAdvancedCache.getComponentRegistry.getComponent(classOf[EntryRetriever[_, _]])
+      val stream = cacheManager.getCache(cacheName).getAdvancedCache.cacheEntrySet.stream().asInstanceOf[CacheStream[_]]
+      segments.map(bitSet => {
+         val segments = bitSet.stream().boxed().collect(Collectors.toSet[Integer])
+         stream.filterKeySegments(segments)
```

Comment:
I would not rely on the fact that distributed streams currently return the same instance.  It could in the future return a new stream so I would recommend to reassign stream to the returned value.